src/utils/response/apiResponse.ts

```diff
@@ -26,6 +26,40 @@ import { Message } from '../../messages/types';
 import { Timezone } from '../../timezones/types';
 import { AbsenceBan } from '../../absenceBans/types';
 
+// Because types cannot be iterated at runtime, we add the keys of Resources here as a value
+// Below we add conditional types that don't compile if this array and Resources go out of sync
+export const resourceNameArray = [
+  'absenceBans',
+  'absenceDays',
+  'absences',
+  'absenceTypes',
+  'changeTimeTrackingRequests',
+  'teamMembers',
+  'departments',
+  'deltaSync',
+  'generalSettings',
+  'todoTasks',
+  'messages',
+  'recentTasks',
+  'projects',
+  'timezones',
+  'tasks',
+  'teams',
+  'serverCommunication',
+  'timePlannings',
+  'timesheetAccountings',
+  'timesheetAccountingSummaries',
+  'timeTrackings',
+  'translations',
+  'favouriteTasks',
+  'users',
+  'usersReadMe',
+  'userStatusOverview',
+  'feedback',
+  'userDefinedFieldDefinitions',
+  'userDefinedFieldDefinitionOptions',
+] as const;
+
```

Comment:
This means for every new one, we have to also maintain this list? Seems not the best solution.

Reply:
Yes. `Resources` is a type, so it's only available at compile time to verify types. So, at runtime we can't get a list of the available resources. That's why `resourceNameArray` is added so we have it as a value. The lines below that make the TS compile fail if this list and the `Resources` type go out of sync, so that it's impossible to add a value to one but not the other